src/layouts/Task/Task.js

```diff
@@ -0,0 +1,172 @@
+  }, [dataUpdateTask?.current,listCategories]);
+  
+  useEffect(() => {
+    if (!cateOfTask) return;
+    dataUpdateTask.current = {
+      ...dataUpdateTask.current,
+      categoryIds: cateOfTask.map((list) => list.value),
+    };
+    handleUpdateTask(dataUpdateTask.current, data.id);
+  }, [cateOfTask]);
+
+  const updateTitleTask = async () => {
+    dataUpdateTask.current = {
+      ...dataUpdateTask.current,
+      title: valueInputTask,
+    };
+    await handleUpdateTask(dataUpdateTask.current, data.id);
+    setEdittask(false);
+  };
+
+  const handleUpdateTask = async (data, id) => {
+    const response = await dispatch(updateTask({ id: id, datatask: data }));
+    if (updateTask.fulfilled.match(response)) {
+      dispatch(getTasks(paramTask));
+    } else {
+      alert("error");
+    }
+  };
+
+  const handleCompleted = async () => {
+    data.status === STATUS.COMPLETED
+      ? (dataUpdateTask.current = {
+          ...dataUpdateTask.current,
+          status: STATUS.IN_PROGRESS,
+        })
+      : (dataUpdateTask.current = {
+          ...dataUpdateTask.current,
+          status: STATUS.COMPLETED,
+        });
+    await handleUpdateTask(dataUpdateTask.current, data.id);
+  };
+
+  const handleDeleteTask = async (id) => {
+    const response = await dispatch(removeTask(id));
+    if (removeTask.fulfilled.match(response)) {
+      await dispatch(getTasks(paramTask));
+    } else {
+      alert("error");
+    }
+  };
+  
+  return (
+    <>
+      <tr>
+        <td>
+          <input
+            type="checkbox"
+            checked={pendingRemoveTasks.includes(data.id)}
+            onChange={() => {
+              reTasks(data.id);
+            }}
+          ></input>
+        </td>
+        <td>
+          <div className="d-flex align-items-center">
+            {!editTask ? (
+              <p>{data?.title}</p>
+            ) : (
+              <>
+                <input
+                  value={valueInputTask}
+                  ref={inputEditRef}
+                  onChange={(e) => {
+                    setValueInputTask(e.target.value);
+                  }}
+                />
+                <button onClick={updateTitleTask}>Update</button>
+              </>
+            )}
+          </div>
+        </td>
+        <td>
+          <Select
+            closeMenuOnSelect={false}
+            defaultValue={defaultCate}
+            isMulti
+            name="colors"
+            options={listcate}
+            className="basic-multi-select"
+            classNamePrefix="select"
+            isDisabled={data.status === "COMPLETED"}
+            onChange={setCateOfTask}
+          />
+        </td>
+        <td> {data?.createdAt}</td>
+        <td>{data?.updatedAt}</td>
+        <td>
+          <input
+            type="checkbox"
+            checked={data.status === "COMPLETED"}
+            onChange={handleCompleted}
+          ></input>
+        </td>
+        <td>
+          {data.status == "IN_PROGRESS" && (
```

Comment:
em lưu ý là mình sẽ luôn dùng so sánh tuyệt đối nha, em sửa tương tự ở các file khác nhé
```suggestion
          {data.status === "IN_PROGRESS" && (
```